addons/purchase/models/account_invoice.py

```diff
@@ -25,7 +25,7 @@ class AccountMove(models.Model):
 
     def _get_invoice_reference(self):
         self.ensure_one()
-        vendor_refs = [ref for ref in set(self.line_ids.mapped('purchase_line_id.order_id.partner_ref')) if ref]
+        vendor_refs = [ref for ref in set(self.invoice_line_ids.mapped('purchase_line_id.order_id.partner_ref')) if ref]
```

Comment:
Hello @smetl just had to change this in 16 as there is no line_ids on self at this point, still ok for you?